complexify-core/src/main/java/com/infinum/complexify/Complexify.kt

```diff
@@ -0,0 +1,101 @@
+package com.infinum.complexify
+
+import android.content.res.Resources
+import com.infinum.android.R
+import com.infinum.complexify.ComplexifyBanMode.*
+import com.infinum.complexify.Constants.CHARSETS_ARRAY
+import com.infinum.complexify.Constants.DEFAULT_BAN_LIST
+import com.infinum.complexify.configuration.Configuration
+import com.infinum.complexify.configuration.DefaultConfiguration
+import kotlin.math.ln
+import kotlin.math.pow
+
+/**
+ * Class for calculation of password complexity.
+ *
+ * @param configuration [Configuration] for password complexity calculation.
+ */
+class Complexify @JvmOverloads constructor(
+    private val configuration: Configuration = DefaultConfiguration()
+) {
+
+    companion object {
+        private const val MIN_COMPLEXITY = 49 // 12 chars with Upper, Lower and Number
+        private const val MAX_COMPLEXITY = 120 // 25 chars, all charsets
+        private const val MAX_PERCENTAGE = 100.0
+    }
```

Comment:
@JonatanPlesko Do these perhaps belong to the DefaultConfiguration?

Reply:
Not really, these are constants used by the current password complexity calculation algorithm. I plan on revamping that one in a later release so I'm not going to spend effort on making this configurable in some way.